Fix docker install

markstuart opened the pull request:
I attempted to install the master (2.9.0a) version of CKAN using docker, but was seeing errors in the logs when running `docker-compose logs -f`. The ckan container would exit almost immediately after starting. 

The only similar info I can see in issues is this comment by @kowh-ai https://github.com/ckan/ckan/issues/5243#issuecomment-595737103

I haven't raised an accompanying issue, I chose to hopefully provide a fix instead.

### Proposed fixes:
[Docker installation instructions for 2.9.0a](https://docs.ckan.org/en/latest/maintaining/installing/install-from-docker-compose.html) should now work as expected.

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

markstuart commented on Dockerfile:
```diff
@@ -44,7 +44,8 @@ RUN useradd -r -u 900 -m -c "ckan account" -d $CKAN_HOME -s /bin/false ckan
     ln -s $CKAN_VENV/bin/pip /usr/local/bin/ckan-pip &&\
-    ln -s $CKAN_VENV/bin/paster /usr/local/bin/ckan-paster
+    ln -s $CKAN_VENV/bin/paster /usr/local/bin/ckan-paster &&\
+    ln -s $CKAN_VENV/bin/ckan /usr/local/bin/ckan
```
I added a symlink to the ckan cli here, similar to what has been done in the past for ckan-paster.

Questions:
* Are we happy with just `ckan`, or maybe `ckan-cli`?
* Do we want to remove the `ckan-paster` symlink?

markstuart commented on Dockerfile:
```diff
@@ -62,4 +63,4 @@ ENTRYPOINT ["/ckan-entrypoint.sh"]
 EXPOSE 5000
 
-CMD ["ckan-paster","serve","/etc/ckan/production.ini"]
+CMD ["ckan","-c","/etc/ckan/production.ini", "run", "--host", "0.0.0.0"]
```
Had to set the host to `0.0.0.0` to allow host network to access the server in the container. The `ckan run` command defaults to `localhost` if `--host` is not provided.

markstuart commented on ckan/cli/__init__.py:
```diff
@@ -5,7 +5,7 @@
 import logging
 from logging.config import fileConfig as loggingFileConfig
-from configparser import ConfigParser
+from six.moves.configparser import ConfigParser
```
Not really sure how this is working for anyone else? This looks to be way that this library is imported in other parts of the project.

markstuart commented on ckan/cli/generate.py:
```diff
@@ -16,6 +16,16 @@
+def extension(output_dir):
     try:
         from cookiecutter.main import cookiecutter
     except ImportError:
```
Since this was in the `@generate` decorator previously, it was also being applied when running `ckan generate config`.

The config command does not need cookiecutter, and should not need to load the dev-requirements in order to generate the config file for a new install.

markstuart commented:
Please note that I'm seeing the issue described in #4715 (`FATAL:  role "postgres" does not exist`) output in the docker logs every 30 seconds or so.

kowh-ai commented:
@smotornyuk -- hey what was the PR mentioned at the Dev meeting today that (may be) updating ckan/cli/generate.py ? I missed that one - thanks Sergey

smotornyuk commented:
#5150 - this one

kowh-ai commented:
@markstuart - thanks for your work here - did you also change the ckan-entrypoint.sh file in your branch?

kowh-ai commented:
Ah - yes I see the 2 changes in the ckan-entrypoint.sh you made - great - this works for me

markstuart commented:
Yep, just updated to use the new ckan CLI commands 👍

markstuart commented:
I see master has moved on a bit since I raised this PR, and there were some conflicts due to @amercader merging #5150. I've rebased my branch on top of master and fixed the conflicts.

kowh-ai commented:
@amercader - Hey can we get this PR merged?...it works fine
